### DISCUSSION TITLE
Consider setting an rlimit for the test suite

### DISCUSSION THREAD

**Discussion by parno** [verus-lang/verus#227] [General]
If the user does not specify an rlimit via the `VERIFY_EXTRA_ARGS` environment variable, then the args we pass to the verifier are instantiated via a `default()` call:
https://github.com/secure-foundations/verus/blob/b44a565193338c90fa50fe2f0620664cf6f26d0f/source/rust_verify/tests/common/mod.rs#L124
This appears to set the `rlimit` to `0`, which means a test can potentially run forever.  I might suggest that we add something like:
```
         our_args.verify_pervasive |= verify_pervasive;
+        our_args.rlimit = DEFAULT_RLIMIT_SECS;
         our_args
```
I tried this as an experiment, and on the CI machine, it causes `e18_pass` from `rust_verify/tests/summer_school.rs` to fail, so we may want to tweak that test or introduce the ability to locally increase the rlimit before adopting a test-suite-wide rlimit.

**Comment by parno** [comment]
We discussed this in today's Verus meeting and the consensus seemed to favor this change.  I might wait to attempt it until after the interpreter branch is merged, since that would be one deterministic way to fix the `e18_pass` test.

**Comment by utaal** [comment]
The test suite now uses Verus' global default (`--rlimit 10`), as we are just calling the regular verus binary for tests (instead of instantiating a verifier in-process).
